internal/cmd/sshauthkeys.go

```diff
@@ -0,0 +1,175 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	logsyslog "log/syslog"
+
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/cmd/cliopts"
+	"github.com/infrahq/infra/internal/linux"
+)
+
+func newSSHDCmd(cli *CLI) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "sshd",
+		Short:  "Commands for integrating with ssh server",
+		Hidden: true,
+	}
+
+	cmd.AddCommand(newSSHDAuthKeysCmd(cli))
+	return cmd
+}
+
+type sshAuthKeysOptions struct {
+	fingerprint    string
+	username       string
+	configFilename string
+}
+
+func newSSHDAuthKeysCmd(cli *CLI) *cobra.Command {
+	var opts sshAuthKeysOptions
+	cmd := &cobra.Command{
+		Use:    "auth-keys",
+		Hidden: true,
+		Args:   ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			logger := setupLogger(cli)
+
+			// sshd_config: AuthorizedKeysCommand infra ssh auth-keys %u %f
+			opts.username = args[0]
+			opts.fingerprint = args[1]
+
+			// log the error to syslog, since we expect stdout/stderr to be hidden
+			if err := runSSHDAuthKeys(cli, logger, opts); err != nil {
+				logger.Err(err).Msg("ssh auth-keys exit")
+				return err
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.configFilename, "config-file",
+		"/etc/infra/connector.yaml", "Path to connector config file")
+	return cmd
+}
+
+func setupLogger(cli *CLI) zerolog.Logger {
+	out := []io.Writer{cli.Stderr}
+	// TODO: log to stderr if this fails?
```

Comment:
writing to a file could be an option too, if all else fails

Reply:
A fallback logfile seems like a good idea